packages/client/src/hooks/useNFTs.js

```diff
@@ -100,9 +139,18 @@ export default function useNFTs() {
     }).catch(console.error);
 
     const collections = (identifiers && identifiers[1]) ?? [];
+
     for (const identifier in collections) {
       await checkCollection(treasuryAddr, collections[identifier]);
+      return;
```

Comment:
what does the return do?

Reply:
return should stop the dispatch to be executed, when we have nfts that need to be added to the state, and the dispatch which is below should reset the collection to be empty in case no nfts are found

Reply:
It was a big brain fart, I just figured too that it will work for the first index only 😆 
I rearranged the code so it is more readable and in a fail-fast manner